src/App.tsx

```diff
@@ -1,24 +1,151 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
+import { Error } from './types/Error';
+import { ErrorMessage } from './components/ErrorMessage';
+import { addTodos, deleteTodos, getTodos } from './api/todos';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
 
-const USER_ID = 0;
+const USER_ID = 27;
+
+export const filterTodos = (currentTodos: Todo[], query: Filter) => {
+  return currentTodos.filter(todo => {
+    switch (query) {
+      case Filter.All:
+        return todo;
+      case Filter.Completed:
+        return todo.completed;
+      case Filter.Active:
+        return !todo.completed;
+      default:
+        return todo;
+    }
+  });
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<Filter>(Filter.All);
+  const [error, setError] = useState<Error | null>(null);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [deleteTodoId, setDeleteTodoId] = useState<number | null>(null);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => setError(Error.UnableToLoadAll));
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    return filterTodos(todos, filter);
+  }, [todos, filter, filterTodos]);
+
+  const activeTodosCount = useMemo(() => {
+    return todos.filter(todo => todo.completed !== true).length;
+  }, [todos]);
+
+  const isCompletedTodos = useMemo(() => {
+    return todos
+      .filter(todo => todo.completed === true).length > 0;
+  }, [todos]);
+
+  const addNewTodo = useCallback((title: string) => {
+    if (title.trim() === '') {
```

Comment:
```suggestion
    if (!title.trim()) {
```

same but a bit shorter